sdk/metric/controller/basic/controller.go

```diff
@@ -0,0 +1,300 @@
+// Copyright The OpenTelemetry Authors
```

Comment:
Compare this file with `sdk/metric/controller/push/push.go`, which shows as deleted. The structure is significantly similar but changes a lot with the addition of error handling and context propagation fixes.